samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -164,8 +205,33 @@ public String getJobModelVersion() {
   }
 
   /**
+   * publish the version number of the next JobModel
+   * @param oldVersion - used to validate, that no one has changed the version in the meanwhile.
+   * @param newVersion - new version.
+   */
+  public void publishNewJobModelVersion(String oldVersion, String newVersion) {
```

Comment:
Same as above: can we rename publishNewJobModelVersion to  updateJobModelVersion??